tablo/models.py

```diff
@@ -268,34 +278,43 @@ def perform_query(self, limit=0, offset=0, **kwargs):
             fields=order_by_fields, related_tables=(None if ids_only else related_tables)
         )
 
+        query_clause = 'SELECT {fields} FROM "{table}" AS "source" {join} {where} {order_by} {limit} {offset}'
+        query_clause = query_clause.format(
+            fields=select_fields, table=self.table, join=join.strip(), where=where.strip(), order_by=order_by,
+            limit='' if limit == 0 else 'LIMIT {limit}'.format(limit=limit + 1),
+            offset='' if offset == 0 else 'OFFSET {offset}'.format(offset=offset)
+        )
+
+        # Execute query with optional limit and offset, and prepare return data
+
         with get_cursor() as c:
-            query_clause = 'SELECT {fields} FROM "{table}" AS "source" {join} {where} {order_by} {limit_offset}'
-            query_clause = query_clause.format(
-                fields=select_fields, table=self.table, join=join.strip(), where=where.strip(), order_by=order_by,
-                limit_offset='LIMIT {limit} OFFSET {offset}'.format(limit=limit, offset=offset)
-            )
             c.execute(query_clause, query_params)
-            data = dictfetchall(c)
+            queried_data = dictfetchall(c)
+
+        limited_data = 0 < limit < len(queried_data)
+        queried_data = queried_data[:-1] if limited_data else queried_data
 
-        return data
+        return {'data': queried_data, 'exceeded_limit': limited_data}
 
     def _alias_fields(self, fields):
-        """ Delimit and alias fields for query in double quotes, but ignore '*' """
+        """ Prepend table alias to fields, delimiting them in double quotes, but ignore '*' """
```

Comment:
In my last commit I updated related field definitions such that instead of using the key `'alias'` to describe the version of a field name that includes the related table (i.e. `'measurements.casgem_station_id'`), we now use `'aliased'`.

The main reason for this change is to comply with the ArcGIS standard, which allows user defined `alias` values per field, even though we don't support this in the context of Tablo. The challenge was to represent qualified field names for related tables in a way that made sense both on the model level (here) and the view level where they are processed and returned.

I considered using `related_name`, `full_name`, `qualified_name` or just `qualified` and a few others but decided on `aliased` because of the following reasons:

1. After searching Postgres documentation, I could find no other term to describe prepending a table alias to a column name
2. In the model we generate SQL with table aliases, apply them to fields, and call this operation `_alias_fields` (this method)
3. We define and reference this version of the field names both at the database and API levels, for instance:

- In `FeatureServiceLayerRelations.fields:766`, which generates the single source of truth for this version of the field name after having queried the database for the related table's column names
- In `FeatureServiceLayer.related_fields:218` where we use the "aliased" version of the field names to prevent name clashes with other related tables when generating related field definitions
- In `interfaces.arcgis.views.convert_wkt_to_esri_feature:409` where we compare "aliased" fields in the query results to this version of the field names to ensure they are grouped correctly

I'm open to feedback, but I think whatever term we use, it should make sense both at the database level (model) and the API level (view). Also, whatever we choose should also fit as part of this method's name so that we avoid the confusion of calling the same operation different things.